src/logger.cpp

```diff
@@ -85,7 +85,11 @@ size_t Logger::errorCount() const
 
 ErrorPtr Logger::getError(size_t index) const
 {
-    return mPimpl->mErrors.at(index);
+    ErrorPtr err = nullptr;
+    if (index < mPimpl->mErrors.size()) {
+        err = mPimpl->mErrors.at(index);
+    }
+    return err;
```

Comment:
What if `index` is lower than zero? Why overcomplicate things? Why not simply have:

```
ErrorPtr Logger::getError(size_t index) const
{
    if ((index >= 0) && (index < mPimpl->mErrors.size())) {
        return mPimpl->mErrors.at(index);
    } else {
        return nullptr;
    }
}
```